.github/workflows/nightly-build.yaml

```
@@ -0,0 +1,46 @@
+name: Nightly build
+
+on:
+  schedule:
+    - cron: '0 0 * * 1-6'
+  workflow_dispatch:
+    inputs: {}
+
+permissions:
+  contents: write
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-nightly:
+    name: Create Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Environment setup
+        uses: ./.github/actions/setup
+
+      - name: Build
+        id: build
+        run: |
+          pnpm build
```

Just to be explicit:

```suggestion
          pnpm build --channel=nightly
```